test/gen-server/lib/homedb/UsersManager.ts

```diff
@@ -0,0 +1,996 @@
+import { User } from 'app/gen-server/entity/User';
+import { AclRuleOrg } from 'app/gen-server/entity/AclRule';
+import { Group } from 'app/gen-server/entity/Group';
+import { Organization } from 'app/gen-server/entity/Organization';
+import { SUPPORT_EMAIL, UsersManager } from 'app/gen-server/lib/homedb/UsersManager';
+import { GetUserOptions, NonGuestGroup, Resource } from 'app/gen-server/lib/homedb/Interfaces';
+import { tmpdir } from 'os';
+import path from 'path';
+import { prepareDatabase } from 'test/server/lib/helpers/PrepareDatabase';
+import { prepareFilesystemDirectoryForTests } from 'test/server/lib/helpers/PrepareFilesystemDirectoryForTests';
+import { EnvironmentSnapshot } from 'test/server/testUtils';
+import { getDatabase } from 'test/testUtils';
+import { Workspace } from 'app/gen-server/entity/Workspace';
+import { Document } from 'app/gen-server/entity/Document';
+import { HomeDBManager } from 'app/gen-server/lib/homedb/HomeDBManager';
+import Sinon, { SinonSandbox, SinonSpy } from 'sinon';
+import { assert } from 'chai';
+import { FullUser, UserProfile } from 'app/common/LoginSessionAPI';
+import { ANONYMOUS_USER_EMAIL, EVERYONE_EMAIL, PREVIEWER_EMAIL, UserOptions } from 'app/common/UserAPI';
+import { Login } from 'app/gen-server/entity/Login';
+import { Pref } from 'app/gen-server/entity/Pref';
+import { EntityManager } from 'typeorm';
+import log from 'app/server/lib/log';
+import winston from 'winston';
+import { updateDb } from 'app/server/lib/dbUtils';
+
+const username = process.env.USER || "nobody";
+const tmpDir = path.join(tmpdir(), `grist_test_${username}_userendpoint`);
+
+describe('UsersManager', function () {
+  this.timeout(30000);
+
+  describe('static method', function () {
+    function* makeIdxIterator() {
+      for (let i = 0; i < 500; i++) {
+        yield i;
+      }
+    }
+
+    function makeUsers(nbUsers: number, idxIt = makeIdxIterator()) {
```

Comment:
Can you use here the `range` function from `lodash`? I think it will look much more simpler, this one is hard to follow.

Reply:
I need to keep track of the increment for users' id across `makeUsers` call. When I call `makeUsers` let's say this way:
```ts
        const idxIt = makeIdxIterator();
        const groupsUsersMap = {
          'editors': makeUsers(2, idxIt),
          'owners': makeUsers(1, idxIt),
        };
```

groupsUsersMap equals to the following object, whose users' id are unique:
```ts
        {
          'editors': [{id: 0, name: 'user 0'}, {id: 1, name: 'user 1'}],
          'owners': [{id: 2, name: 'user 2'}]
        }
```

I don't know how to make a similar thing with `lodash.range()`.

An alternative I can do, but I find it less elegant, would consist in passing the start index each time:
```ts
let startIdx = 0;
const groupsUsersMap = {};
groupsUsersMap['editors'] = makeUsers(2, startIdx);
startIdx += 2;
groupsUsersMap['owners'] = makeUsers(1, startIdx);
startIdx += 1;
```